librz/core/gdiff.c

```diff
@@ -34,8 +39,53 @@ RZ_API int rz_core_gdiff_fcn(RzCore *c, ut64 addr, ut64 addr2) {
 	return true;
 }
 
-/* Fingerprint functions and blocks, then diff. */
-RZ_API int rz_core_gdiff(RzCore *c, RzCore *c2) {
+/**
+ * \brief Calculates basic block differences of 2 functions within 2 files
+ *
+ * Calculates basic block differences of 2 functions within 2 files
+ * */
+RZ_API bool rz_core_gdiff_function_2_files(RzCore *c, RzCore *c2, ut64 addr, ut64 addr2) {
+	RzList *la, *lb;
+	RzAnalysisFunction *fa = rz_analysis_get_function_at(c->analysis, addr);
+	RzAnalysisFunction *fb = rz_analysis_get_function_at(c2->analysis, addr2);
+	if (!fa || !fb) {
+		eprintf("cannot get functions at 0x%" PFMT64x " or at 0x%" PFMT64x "\n", addr, addr2);
+		return false;
+	}
+	RzAnalysisBlock *bb;
+	RzListIter *iter;
+	rz_list_foreach (fa->bbs, iter, bb) {
+		if (rz_analysis_diff_fingerprint_bb(c->analysis, bb) < 0) {
+			eprintf("cannot fingerprint 0x%" PFMT64x "\n", addr);
+			return false;
+		}
+	}
+	rz_list_foreach (fb->bbs, iter, bb) {
+		if (rz_analysis_diff_fingerprint_bb(c2->analysis, bb) < 0) {
+			eprintf("cannot fingerprint 0x%" PFMT64x "\n", addr2);
+			return false;
+		}
+	}
+
+	rz_analysis_diff_fingerprint_fcn(c->analysis, fa);
+	rz_analysis_diff_fingerprint_fcn(c2->analysis, fb);
+
+	la = rz_list_new();
+	rz_list_append(la, fa);
+	lb = rz_list_new();
+	rz_list_append(lb, fb);
+	rz_analysis_diff_fcn(c->analysis, la, lb);
+	rz_list_free(la);
+	rz_list_free(lb);
+	return true;
+}
+
+/**
+ * \brief Calculates basic block differences of all functions within 2 files
+ *
+ * Calculates basic block differences of all functions within 2 files.
+ * */
+RZ_API bool rz_core_gdiff(RzCore *c, RzCore *c2) {
```

Comment:
Maybe name it accordingly, `rz_core_gdiff_2_files()` like you did with other functions?